main.go

```diff
@@ -121,6 +123,17 @@ func main() {
 
 	s := scheduler.NewScheduler(log.Log.WithValues("component", "scheduler"))
 
+	// TODO: take this variable from helm env
+	upgradeCheckingDisabled := true
+	if !upgradeCheckingDisabled {
+		setupLog.Info("upgrade checking enabled")
+
+		err = upgradecheck.StartUpgradeCheckJob(context.TODO(), mgr.GetClient(), s)
```

Comment:
1. upgradecheck.StartUpgradeCheckJob -> upgradecheck.Start()
2. do you actually need a context?

Reply:
There are can be other jobs in this package, so I would like to keep it.
Context we need for logging and getting operator deployment